aodntools/timeseries_products/common.py

```diff
@@ -6,66 +6,70 @@ class NoInputFilesError(Exception):
     pass
 
 
-def check_file(nc, site_code, VoI):
+def check_file(nc, site_code, variables_of_interest):
     """
-    Return list of errors found in the file.
+    Check that a file meets the requirements for inclusion in a product.
+    Return a list of errors
+    
     Checks applied:
     * Correct site_code
-    * Variables of interest is present
-    * Variables TIME, LATITUDE and LONGITUDE are present
-    * NOMINAL_DEPTH is present as variable or attribute
     * file_version is FV01
-    * if LATITUDE or LONIGUTDE are dimension, they have length 1
+    * Coordinate variables TIME, LATITUDE and LONGITUDE are present
+    * NOMINAL_DEPTH is present as variable or attribute (instrument_nominal_depth)
+    * At least one variable of interest is present
+    * All variables of interest have only the allowed dimensions 
+    * If LATITUDE or LONIGUTDE are dimension, they have length 1
     * Global attributes time_deployment_start and time_deployment_end exist
 
-    :param nc: xarray dataset
+    :param nc: open xarray dataset
     :param site_code: code of the mooring site
-    :param VoI: string. Variable of Interest
+    :param variables_of_interest: variable name or list of names to be included in the product
     :return: list of failed tests
     """
 
+    if isinstance(variables_of_interest, str):
+        variables_of_interest = [variables_of_interest]
     attributes = list(nc.attrs)
     variables = list(nc.variables)
-    allowed_dimensions = ['TIME', 'LATITUDE', 'LONGITUDE']
+    allowed_dimensions = {'TIME', 'LATITUDE', 'LONGITUDE'}
```

Comment:
I don't understand these definitions here. If you wish to maintain them at the function level, put them close to their use. If they are constants that are used elsewhere, define them as constants outside for re-use.

Reply:
yeap, or that.